kernel/rtlil.cc

```diff
@@ -445,6 +445,34 @@ vector<string> RTLIL::AttrObject::get_hdlname_attribute() const
 	return split_tokens(get_string_attribute(ID::hdlname), " ");
 }
 
+void RTLIL::AttrObject::set_intvec_attribute(RTLIL::IdString id, const vector<int> &data)
+{
+	std::stringstream attrval;
+	for (auto &i : data) {
+		if (attrval.tellp() > 0)
+			attrval << " ";
+		attrval << i;
+	}
+	attributes[id] = RTLIL::Const(attrval.str());
+}
+
+vector<int> RTLIL::AttrObject::get_intvec_attribute(RTLIL::IdString id) const
+{
+	vector<int> data;
+	auto it = attributes.find(id);
+	if (it != attributes.end())
+		for (const auto &s : split_tokens(attributes.at(id).decode_string())) {
+			char *end = nullptr;
+			errno = 0;
+			long value = strtol(s.c_str(), &end, 10);
+			log_assert(end == s.c_str() + s.size());
+			if (errno == ERANGE || value < INT_MIN || value > INT_MAX)
+				log_error("Literal for intvec attribute is out of range");
+			data.push_back(value);
```

Comment:
Because you used `log_error` rather `log_cmd_error`, this line would go ahead and invoke UB anyway for out-of-range literals. The errors should be `log_cmd_error`, which throw an exception.

Reply:
Woops, fixed.

Reply:
Looks good now.